Wikipedia/Code/WelcomeAnimationExtensions.swift

```diff
@@ -0,0 +1,753 @@
+import Foundation
+
+struct WelcomePlus {
+    var position: CGPoint
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeCircle {
+    var position: CGPoint
+    var radius: CGFloat
+    var isDashed: Bool
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeLine {
+    var position: CGPoint
+    var width: CGFloat
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeImage {
+    var name: String
+    var frame: CGRect
+    var zPosition: CGFloat
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeBar {
+    var frame: CGRect
+    var color: UIColor
+    var cornerRadius: CGFloat
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeAnimation {
```

Comment:
This struct is essentially just the animation parameters… so we can just create an animation with these values instead. 

Structs are nice, but if you find yourself doing a 1-1 mapping between a class's properties, its best just to just use the class.